iml-services/iml-action-runner/src/data.rs

```diff
@@ -98,6 +100,100 @@ pub(crate) async fn await_next_session(
+) -> Result<HashMap<String, Vec<LdevEntry>>, ActionRunnerError> {
+    tracing::debug!("getting hosts");
+    let hosts = sqlx::query!("SELECT id, fqdn FROM chroma_core_managedhost WHERE not_deleted=True")
+        .fetch_all(&pool)
+        .await?
+        .into_iter()
+        .map(|x| (x.id, x.fqdn))
+        .collect::<HashMap<i32, String>>();
+
+    tracing::debug!("hosts: {:?}", hosts);
+    let fs_to_fqdn_map = db::get_mgs_host_fqdn(&pool).await?;
+
+    tracing::debug!("fs_to_fqdn_map: {:?}", fs_to_fqdn_map);
+    // Get devices from the `targets` endpoint
+    let client = iml_manager_client::get_client()?;
+    let targets: Vec<Target> = iml_manager_client::graphql(
+        client,
+        target::query::build(false, None::<String>, None, None, None),
+    )
+    .await?;
+
+    tracing::debug!("targets: {:?}", targets);
+
+    let ldev_entries = targets
+        .into_iter()
+        .filter_map(|x| {
+            if let Some(primary) = x.active_host_id {
+                if let Some(mount_path) = x.mount_path {
+                    let failovers = x
+                        .host_ids
+                        .into_iter()
+                        .filter(|x| *x != primary)
+                        .filter_map(|x| hosts.get(&x))
+                        .map(|x| x.to_string())
+                        .collect::<Vec<String>>();
+
+                    let failover = if failovers.is_empty() {
+                        None
+                    } else {
+                        Some(failovers[0].to_string())
+                    };
+
+                    let ldev_entry = LdevEntry {
+                        primary: hosts
+                            .get(&primary)
+                            .expect("Couldn't get primary host.")
+                            .to_string(),
+                        failover,
+                        label: x.name,
+                        device: mount_path,
+                    };
+
+                    return Some((
+                        fs_to_fqdn_map
+                            .get(&x.filesystems[0])
+                            .expect("get first filesystem"),
+                        ldev_entry,
+                    ));
+                } else {
+                    panic!("All targets must be mounted when configuring ldev conf.");
+                }
+            } else {
+                panic!("All targets must be mounted on an active host when configuring ldev conf.");
+            }
+        })
+        .fold(
+            HashMap::<String, Vec<LdevEntry>>::new(),
+            |mut acc, (fqdn, ldev_entry)| {
+                let items = acc.entry(fqdn.to_string()).or_insert(vec![]);
+                (*items).push(ldev_entry);
+
+                acc
+            },
+        );
+
+    tracing::debug!("ldev entries: {:?}", ldev_entries);
+
+    Ok(ldev_entries)
+}
+
+#[derive(serde::Deserialize, Debug, Eq, PartialEq, Ord, PartialOrd, Clone)]
+#[serde(rename_all="camelCase")]
```

Comment:
You should not need this with mappings like `creared_at: crearedAt` in GraphQL queries.